internal/docker/container.go

```diff
@@ -59,89 +78,124 @@ func (r *ContainerRunner) pullImage(img string) error {
 	return nil
 }
 
-// setup performs any necessary steps for a test runner to execute tests.
-func (r *ContainerRunner) setupImage(confd config.Docker, beforeExec []string, project interface{}, files []string) error {
+// fetchImage ensure that container image is available for test runner to execute tests.
+func (r *ContainerRunner) fetchImage(docker *config.Docker) error {
 	if !r.docker.IsInstalled() {
 		return fmt.Errorf("please verify that docker is installed and running: " +
 			" follow the guide at https://docs.docker.com/get-docker/")
 	}
 
-	if confd.Image == "" {
+	if docker.Image == "" {
 		img, err := r.ImageLoc.GetImage(r.Ctx, r.Framework)
 		if err != nil {
 			return fmt.Errorf("unable to determine which docker image to run: %w", err)
 		}
-		confd.Image = img
+		docker.Image = img
 	}
 
-	if err := r.pullImage(confd.Image); err != nil {
+	if err := r.pullImage(docker.Image); err != nil {
 		return err
 	}
+	return nil
+}
 
-	container, err := r.docker.StartContainer(r.Ctx, files, confd)
+func (r *ContainerRunner) startContainer(options containerStartOptions) (string, error) {
+	container, err := r.docker.StartContainer(r.Ctx, options)
 	if err != nil {
-		return err
+		return "", err
 	}
-	r.containerID = container.ID
+	containerID := container.ID
 
-	pDir, err := r.docker.ProjectDir(r.Ctx, confd.Image)
+	pDir, err := r.docker.ProjectDir(r.Ctx, options.Docker.Image)
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	r.containerConfig.jobDetailsFilePath, err = r.docker.JobDetailsURLFile(r.Ctx, options.Docker.Image)
+	if err != nil {
+		return "", err
 	}
 
 	tmpDir, err := ioutil.TempDir("", "saucectl")
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer os.RemoveAll(tmpDir)
 
 	rcPath := filepath.Join(tmpDir, SauceRunnerConfigFile)
-	if err := jsonio.WriteFile(rcPath, project); err != nil {
-		return err
+	if err := jsonio.WriteFile(rcPath, options.Project); err != nil {
+		return "", err
 	}
 
-	if err := r.docker.CopyToContainer(r.Ctx, r.containerID, rcPath, pDir); err != nil {
-		return err
+	if err := r.docker.CopyToContainer(r.Ctx, containerID, rcPath, pDir); err != nil {
+		return "", err
 	}
 	r.containerConfig.sauceRunnerConfigPath = path.Join(pDir, SauceRunnerConfigFile)
 
 	// running pre-exec tasks
-	err = r.beforeExec(beforeExec)
+	err = r.beforeExec(containerID, options.SuiteName, options.BeforeExec)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return nil
+	return container.ID, nil
 }
 
-func (r *ContainerRunner) run(cmd []string, env map[string]string) error {
+func (r *ContainerRunner) run(containerID, suiteName string, cmd []string, env map[string]string) (string, string, string, bool, error) {
 	defer func() {
-		log.Info().Msg("Tearing down environment")
-		if err := r.docker.Teardown(r.Ctx, r.containerID); err != nil {
+		log.Info().Msgf("%s: Tearing down environment", suiteName)
```

Comment:
Might as well make use of our beautiful structured logger 💪 

```suggestion
		log.Info().Str("suite", suiteName).Msg("Tearing down environment", suiteName)
```